GEOSwift/GEOS/Envelope.swift

```diff
@@ -0,0 +1,48 @@
+public struct Envelope: Hashable {
+    public internal(set) var minX: Double
+    public internal(set) var maxX: Double
+    public internal(set) var minY: Double
+    public internal(set) var maxY: Double
+
+    public init(minX: Double, maxX: Double, minY: Double, maxY: Double) {
+        precondition(minX <= maxX, "Envelope: minX must not be greater than maxX")
+        precondition(minY <= maxY, "Envelope: minY must not be greater than maxY")
+        self.minX = minX
+        self.maxX = maxX
+        self.minY = minY
+        self.maxY = maxY
+    }
+
+    public var minXMaxY: Point {
+        return Point(x: minX, y: maxY)
+    }
+
+    public var maxXMaxY: Point {
+        return Point(x: maxX, y: maxY)
+    }
+
+    public var minXMinY: Point {
+        return Point(x: minX, y: minY)
+    }
+
+    public var maxXMinY: Point {
+        return Point(x: maxX, y: minY)
+    }
+}
+
+extension Envelope: GeometryConvertible {
+    public var geometry: Geometry {
+        if minX == maxX, minY == maxY {
+            return .point(Point(x: minX, y: minY))
+        } else {
+            // swiftlint:disable:next force_try
+            return try! .polygon(Polygon(exterior: Polygon.LinearRing(
```

Comment:
Why is this safe to unwrap?

Reply:
Looks like it could only throw if a point has less than 2 coordinates, and it's hard coded to have the right amount here.

Reply:
There are a few different things happening in this expression:

1. First, it creates 5 points from `minX`, `minY`, `maxX`, `maxY`. `Point.init(x:y:)` cannot throw.
1. Next, It creates a linear ring from those 5 points. `Polygon.LinearRing.init(points:)` only throws if the number of points is not greater than or equal to 4 or if the first and last points are not equivalent. ([link](https://github.com/GEOSwift/GEOSwift/pull/167/files/40344d812e752ccd8f1ad7f3b383a8e160dc0076#diff-82e60e42d2a1453ac86c22c430d4c4d8R14)). In this case, we know for sure that we meet those criteria, so that part is also safe.
1. Next, we create a polygon from the linear ring. That cannot throw.
1. Last, we wrap the polygon in `Geometry.polygon`. That also cannot throw.

and that's it, so that's why it's safe to force try